tests/scripts/thread-cert/v1_2_test_parent_selection.py

```diff
@@ -0,0 +1,309 @@
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import unittest
+
+import thread_cert
+import mle
+
+LEADER_1_2 = 1
+ROUTER_1_1 = 2
+REED_1_2 = 3
+ROUTER_1_2 = 4
+REED_1_1 = 5
+MED_1_1 = 6
+MED_1_2 = 7
+
+# Topology
+#               (lq:2)  (pp:1)
+#     REED_1_2  ----- ROUTER_1_2
+#        |     \    /     |      \
+#        |       \/    REED_1_1    \
+# (lq:2) |      /  \  /  `router`    \
+#        | (lq:2)    \                 \
+#        |  /      /   \                 \
+#       LEADER_1_2 --- ROUTER_1_1 -- MED_1_2
+#                \        |
+#                  \      |
+#                    \    |
+#                       MED_1_1
+#
+# 1) bring up LEADER_1_2 and ROUTER_1_1,
+# 2) config lq(LEADER_1_2->REED_1_2) as 2, bring up REED_1_2 would attach to ROUTER_1_1
+#    due to higher two-way link quality,
+# 3) config lq(LEADER_1_2->ROUTER_1_2) and lq(REED_1_2->ROUTER_1_2) as 2, ROUTER_1_2 would attach to LEADER_1_2
+#    due to active router is preferred,
+# 4) config parent priority as 1 for ROUTER_1_2, REED_1_1 would attach to ROUTER_1_2
+#    due to higher parent priority,
+# 5) upgrade REED_1_1 to `router` role, MED_1_1 would attach to LEADER_1_2 which has higher link quality 3 field
+# 6) connfig parent priority as 1 for ROUTER_1_1, MED_1_2 would attach to ROUTER_1_2 due to higher version
+#
+
+
+class TestParentSelection(thread_cert.TestCase):
+    topology = {
+        LEADER_1_2: {
+            'version': '1.2',
+            'whitelist': [REED_1_2, ROUTER_1_2, REED_1_1, ROUTER_1_1, MED_1_1],
+        },
+        ROUTER_1_1: {
+            'version': '1.1',
+            'whitelist': [LEADER_1_2, REED_1_2, MED_1_2, MED_1_1],
+        },
+        REED_1_2: {
+            'version': '1.2',
+            'whitelist': [ROUTER_1_2, ROUTER_1_1, LEADER_1_2],
+        },
+        ROUTER_1_2: {
+            'version': '1.2',
+            'whitelist': [REED_1_2, MED_1_2, REED_1_1, LEADER_1_2],
+        },
+        REED_1_1: {
+            'version': '1.1',
+            'whitelist': [ROUTER_1_2, LEADER_1_2]
+        },
+        MED_1_1: {
+            'mode': 'rs',
+            'version': '1.1',
+            'whitelist': [LEADER_1_2, ROUTER_1_1],
+        },
+        MED_1_2: {
+            'mode': 'rs',
+            'version': '1.2',
+            'whitelist': [ROUTER_1_1, ROUTER_1_2],
+        },
+    }
+    """All nodes are created with default configurations"""
+
+    def test(self):
+
+        self.nodes[LEADER_1_2].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER_1_2].get_state(), 'leader')
+
+        self.nodes[ROUTER_1_1].set_router_selection_jitter(1)
+        self.nodes[ROUTER_1_1].start()
+        self.simulator.go(5)
```

Comment:
I've seen 7 seconds allowance in [other tests](https://github.com/openthread/openthread/blob/da4e64805219e695962b96f570244907e72f0b4a/tests/scripts/thread-cert/v1_2_router_5_1_1.py#L55) for router attachment. Should we define these timings globally?

Reply:
Agree it would be better. May check through the other tests to see if there are any case-specific requirements and do the updates in a separate PR.